instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/rpc/MetricsView.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.rpc;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+// this is temporary, see
+// https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3962#issuecomment-906606325
+@SuppressWarnings("rawtypes")
+final class MetricsView {
+
+  private static final Set<AttributeKey> recommended = buildRecommended();
+  private static final Set<AttributeKey> optional = buildOptional();
+
+  private static Set<AttributeKey> buildRecommended() {
+    // the list of recommended metrics attributes is from
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#attributes
+    Set<AttributeKey> view = new HashSet<>();
+    view.add(SemanticAttributes.RPC_SYSTEM);
+    view.add(SemanticAttributes.RPC_SERVICE);
+    view.add(SemanticAttributes.RPC_METHOD);
+    return view;
+  }
+
+  private static Set<AttributeKey> buildOptional() {
+    // the list of optional metrics attributes is from
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/rpc.md#attributes
+    Set<AttributeKey> view = new HashSet<>();
+    view.add(SemanticAttributes.NET_PEER_IP);
+    view.add(SemanticAttributes.NET_PEER_NAME);
+    view.add(SemanticAttributes.NET_PEER_PORT);
+    view.add(SemanticAttributes.NET_TRANSPORT);
+    return view;
```

Comment:
According to the linked doc:

```
To avoid high cardinality, implementations should prefer the most stable of net.peer.name or net.peer.ip, depending on expected deployment profile. For many cloud applications, this is likely net.peer.name as names can be recycled even across re-instantiation of a server with a different ip.
```

I.e. you can *fallback* to net.peer.ip if it's you're only option, but you should JUST use net.peer.name if you can.

Reply:
@jsuereth so we should go ahead and capture these same attributes on server metrics (unlike on http server metrics)? I guess the situation is not as bad in RPC-land as in browser-land (except for maybe android/grpc?)

Reply:
We may want to re-evaluate server side using a notion of endpoint vs. net.peer.ip for similar reasons to http.  However the specification is based on what gRPC does.  I'll do some homework here to see if this is still how that's instrumented.

Key thing though, we should be using hostname over IP because hostname should be more stable, especially in the presence of a load balancer.

Reply:
I understanding is  usering net.peer.name instead of net.peer.ip if the attribute  net.peer.name exist.

Reply:
Yeah, the other issue here is around whether we *should* be using `net.peer.*` on RPC server-metrics.   

We translated the specification on instrumentation from OpenCensus gRPC that lead to this as a basis.  HOWEVER, there's a few caveats:

1. In OpenCensus instrumentation + Views were separately specified.  Instrumentation was thorough, views were more conservative.   When this was originally written for OTEL the View spec wasn't known, so that part is unspecified.   In OpenCensus, views are where the high cardinality issues were dealt with by use case, and where you could select styles of latency metrics to export (e.g. if you wanted SLO monitoring).   I think the balance we struck 13 months ago is likely now wrong given the evolution of our APIs/SDKs *unless* we provide out-of-the-box VIew configuration for these metrics in addition to the rich instrumentation attributes.
2. RPC semantic conventions now diverge from HTTP semantic conventions.  Effectively....   "client" metrics use `net.peer.name` and "host" metrics should use `host.name` in HTTP, but RPC both use `net.peer`.

For both of these reasons, we may want to re-evaluate RPC metrics.  Specifically RPC server metrics.

Reply:
Opened https://github.com/open-telemetry/opentelemetry-specification/issues/2214

Reply:
So I can modify the RPC conventions implement like this:
Client metrics use net.peer.name (or if unavailable net.peer.ip),net.peer.port,net.transport.
Server metrics use net.host.name (or if unavailable nethost.ip),net.transport,exclude net.host.port.
Do you think this is ok?

Reply:
That's effectively what I'd do.

For the purpose of this PR, I think if you just don't fill out any `net.*` attributes on the server side you're fine (because of Resource attributes).   As we discussed in the Java-SiG yesterday, it'd be good to mark this instrumentation as "experimental" in some fashion so you can try a few options out and we can see how they work in practice.